### ISSUE TITLE
SDK Generation for Microsoft.StorageSync RP created files in Storage folder instead of StorageSync folder

### ISSUE THREAD

**Issue by ankushbindlish2** [Azure/azure-sdk-for-net#4681]
1. Classes were created in H:\csharpsdk\azure-sdk-for-net\src\SDKs\Storage\Management.StorageSync
2. Log file is empty : H:\csharpsdk\azure-sdk-for-net\src\SDKs\_metadata\storagesync_resource-manager.txt
2. StorageSyncService class was inherited from TrackedResource class in swagger spec but in generation it was derived from IResource., and hence failed compiling.
Actual : public partial class StorageSyncService : IResource
Expected : public partial class StorageSyncService : TrackedResource

Swagger spec : https://github.com/Azure/azure-rest-api-specs/blob/master/specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-04-02/storagesync.json
Sdk fork : https://github.com/ankushbindlish2/azure-sdk-for-net
Branch : https://github.com/ankushbindlish2/azure-sdk-for-net/tree/storagesync-addsdk
Local share : 
\\ankushb-dt\csharpsdk\azure-rest-api-specs
\\ankushb-dt\csharpsdk\azure-sdk-for-net\src\SDKs\StorageSync

 "StorageSyncService": {
      "type": "object",
      "description": "Storage Sync Service object.",
      "allOf": [
        {
          "$ref": "#/definitions/TrackedResource"
        },
        {
          "properties": {
            "properties": {
              "x-ms-client-flatten": true,
              "$ref": "#/definitions/StorageSyncServiceProperties",
              "description": "Storage Sync Service properties."
            }
          }
        }
      ],
      "x-ms-azure-resource": true
    },

PS H:\csharpsdk\azure-sdk-for-net\src\SDKs\StorageSync> dir

    Directory: H:\csharpsdk\azure-sdk-for-net\src\SDKs\StorageSync

Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----        8/20/2018  11:54 AM                Management.StorageSync
d-----        8/20/2018  11:55 AM                StorageSync.Tests

PS H:\csharpsdk\azure-sdk-for-net\src\SDKs\StorageSync> cd .\Management.StorageSync\
PS H:\csharpsdk\azure-sdk-for-net\src\SDKs\StorageSync\Management.StorageSync> .\generate.ps1

    Directory: H:\csharpsdk\azure-sdk-for-net\src\SDKs\_metadata

Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----        8/20/2018  11:59 AM              0 storagesync_resource-manager.txt
Installing AutoRest version: latest

> autorest@2.0.4283 preinstall C:\Users\ankushb\AppData\Roaming\npm\node_modules\autorest
> node ./preinstall-check

C:\Users\ankushb\AppData\Roaming\npm\autorest -> C:\Users\ankushb\AppData\Roaming\npm\node_modules\autorest\dist\app.js
+ autorest@2.0.4283
updated 1 package in 1.63s

AutoRest installed successfully.
Commencing code generation
Generating CSharp code
Executing AutoRest command
cmd.exe /c autorest.cmd https://github.com/Azure/azure-rest-api-specs/blob/master/specification/storagesync/resource-manager/readme.md --csharp --version=latest --reflect-api-versions --csharp-sdks-folder=H:\csharpsdk\azure-sdk-for-net\src\SDKs
AutoRest code generation utility [version: 2.0.4283; node: v8.11.2]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
C:\Users\ankushb\.autorest\@microsoft.azure_autorest-core@2.0.4283\autorest-core -> C:\Users\ankushb\.autorest\@microsoft.azure_autorest-core@2.0.4283\node_modules\@microsoft.azure\autorest-core\dist\app.js
C:\Users\ankushb\.autorest\@microsoft.azure_autorest-core@2.0.4283\autorest-language-service -> C:\Users\ankushb\.autorest\@microsoft.azure_autorest-core@2.0.4283\node_modules\@microsoft.azure\autorest-core\dist\language-service\language-service.js
   Loading AutoRest core      'C:\Users\ankushb\.autorest\@microsoft.azure_autorest-core@2.0.4283\node_modules\@microsoft.azure\autorest-core\dist' (2.0.4283)
   Installing AutoRest extension '@microsoft.azure/autorest.csharp' (~2.3.79)
   Installing AutoRest extension '@microsoft.azure/autorest.modeler' (2.3.55)

2018-08-20 19:00:27 UTC
Azure-rest-api-specs repository information
GitHub fork: Azure
Branch:      master
Commit:      ce519b0f5ae661b74635ed658e0ec684e78bf5a8
AutoRest information
Requested version: latest
Bootstrapper version:    autorest@2.0.4283

Log file can be found at location H:\csharpsdk\azure-sdk-for-net\src\SDKs\_metadata\storagesync_resource-manager.txt

PS H:\csharpsdk\azure-sdk-for-net\src\SDKs\StorageSync\Management.StorageSync>
PS H:\csharpsdk\azure-sdk-for-net\src\SDKs\StorageSync\Management.StorageSync> H:\csharpsdk\azure-sdk-for-net\src\SDKs\_metadata\storagesync_resource-manager.txt

**Comment by shahabhijeet** [comment]
@ankushbindlish2 regarding the compiler error you are seeing in the generated code, please model the TrackedResource using the common model definition from this location
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v1/types.json#L45
Here is one such example of it's usage
https://github.com/Azure/azure-rest-api-specs/blob/ac8857e5386be4f11d7ce84183ec1e686f585af4/specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/dscConfiguration.json#L656

**Comment by ankushbindlish2** [comment]
Sure, Will try.

1. What's the problem with underlying swagger spec. I have seen similar spec with other RP.
2. Did we find out why we are generating it in wrong location. storage instead of storagesync

**Comment by dsgouda** [comment]
@ankushbindlish2 Are you planning to create a new directory under src/SDKs/StorageSync or do you wish to generate the code alongside src/SDKs/Storage/Microsoft.Storage?
For case 1, please modify the config file [here](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/storagesync/resource-manager/readme.md) and modify the output-folder under C# section (replace SDKs/Storage with SDKs/StorageSync)
For case 2, I believe the files should already be generated as expected.
Let us know if there is anything else we can help with

**Comment by ankushbindlish2** [comment]
Abhijeet Shah <Abhijeet.Shah@microsoft.com> has all context on this issue.

yes. StorageSync, Will look at case 1

**Comment by ankushbindlish2** [comment]
will fix read me.  output-folder: $(csharp-sdks-folder)/Storage/Management.StorageSync/Generated

**Comment by ankushbindlish2** [comment]
thanks !

**Comment by dsgouda** [comment]
@ankushbindlish2 merged the REST spec PR. Please try regenerating and close the issue if you're unblocked

**Comment by ankushbindlish2** [comment]
Sure. Will try first thing tomorrow morning.

**Comment by ankushbindlish2** [comment]
It solved issue 1 , but not 2 and 3

**Comment by dsgouda** [comment]
@ankushbindlish2 could you explain what the problem is now? Feel free to ping offline

**Comment by dsgouda** [comment]
@ankushbindlish2 dp you still have any issues here

**Comment by ankushbindlish2** [comment]
no. I will continue prepare a c# sdk pull request.

**Comment by dsgouda** [comment]
Perfect, feel free to open a new issue if anything is off
